auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -515,23 +543,18 @@ public SignInIntentBuilder setAllowNewEmailAccounts(boolean enabled) {
             return this;
         }
 
-        private boolean isIdpAlreadyConfigured(@NonNull String providerId) {
-            for (IdpConfig config : mProviders) {
-                if (config.getProviderId().equals(providerId)) {
-                    return true;
-                }
-            }
-            return false;
-        }
-
         public Intent build() {
             return KickoffActivity.createIntent(mApp.getApplicationContext(), getFlowParams());
         }
 
         @VisibleForTesting()
         public FlowParameters getFlowParams() {
+            if (mProviders.isEmpty()) {
+                mProviders.add(new IdpConfig.Builder(EMAIL_PROVIDER).build());
+            }
```

Comment:
This is why we should always do this kind of thing in the `build` method: if people call `setProviders(Collections.emptyList())`, we will start the auth flow with no providers. Now, we can guarantee that at least the email provider is provided.